Cloud docker 1.0.0 (#6504)
magento/devdocs: src/cloud/release-notes/backward-incompatible-changes.md

### Service version requirement changes

We changed the minimum PHP version requirement from 7.0.x to 7.1.x for Cloud projects that use `{{ site.data.var.ct }}` v2002.1.0 and later. If your environment configuration specifies PHP 7.0, update the [php configuration]({{ site.baseurl }}/cloud/project/project-conf-files_magento-app.html#configure-php-options) in the `.magento.app.yaml` file.

### Environment configuration changes

The following table provides information about environment variables and other environment configuration files that were removed or deprecated in `{{ site.data.var.ct }}` v2002.1.0.

   Item | Replacement
   -------- | -------
   `SCD_EXCLUDE_THEMES` variable | [`SCD_MATRIX`]({{ site.baseurl}}/cloud/env/variables-build.html#scd_matrix)
   `STATIC_CONTENT_THREADS` variable | [`SCD_THREADS`]({{ site.baseurl}}/cloud/env/variables-build.html#scd_threads)
   `DO_DEPLOY_STATIC_CONTENT` variable | [`SKIP_SCD`]({{ site.baseurl}}/cloud/env/variables-build.html#skip_scd)
   `STATIC_CONTENT_SYMLINK` variable | None. Now, the build always creates a symlink to the static content directory `pub/static`.

>    `build_options.ini` file | Use the [`.magento.env.yaml`]({{ site.baseurl }}/cloud/project/magento-env-yaml.html)) file to configure environment variables to manage build and deploy actions across all your environments.<br><br>If you build a Cloud environment that includes the `build_options.ini` file, the build fails.

Reviewer: There's an extra paren.

```suggestion
   `build_options.ini` file | Use the [`.magento.env.yaml`]({{ site.baseurl }}/cloud/project/magento-env-yaml.html) file to configure environment variables to manage build and deploy actions across all your environments.<br><br>If you build a Cloud environment that includes the `build_options.ini` file, the build fails.
```

-  **Updating the apply patches command**–We moved the command for applying patches to Magento Cloud from the `vendor/bin/ece-tools` directory to the `vendor/bin/ece-patches` directory. If you use this command to apply patches manually, use the new path.

   > Manually apply patches

   ```bash
   php ./vendor/bin/ece-patches apply
   ```

## Magento Cloud Docker changes

-  **The minimum PHP version requirement is now PHP 7.1**–If your Magento Cloud Docker host is running an earlier version, upgrade to PHP v7.1 or later.

-  **Magento Cloud Docker command changes**–

   -  **Updating Magento Cloud Docker commands for Docker build operations**–We moved the Magento Cloud Docker commands from the `vendor/bin/ece-tools` directory to the `vendor/bin/ece-docker` directory. Update your scripts and commands to use the new path.

      After upgrading to the {{site.data.var.ct}} 2002.1.0 or {{site.data.var.mcd}} 1.0.0, use the following command to view available `ece-docker` commands.

      ```bash
      php ./vendor/bin/ece-docker list
      ```

>    -  **Updating the Magento Cloud docker-compose commands**–We changed the name of the `./bin/docker` directory to `./bin/magento-docker`.  Update your scripts and commands to use the new path.

Reviewer 2: The directory didn't change, but the path did. We could say it was renamed instead.
```suggestion
   -  **Updating the Magento Cloud docker-compose commands**–We renamed `./bin/docker` to `./bin/magento-docker`.  Update your scripts and commands to use the new path.
```

   -  **Using temporary containers**–In previous versions, the containers created by `bin/magento-docker` command operations were not removed, so you could use them for other operations.  Now, the `magento-docker` commands remove any containers they create after the command completes.

      If you want to keep a container created by a docker-compose operation, use the `docker-compose run` command instead of the `bin/magento-docker` command.

   -  **Running post-deploy hooks**–The `cloud-deploy` command no longer runs post deploy hooks. You must use the new `cloud-post-deploy` command to run post deploy hooks after you deploy.

      ```bash
      bin/magento-docker ece-deploy
      ```

      ```bash
      bin/magento-docker ece-post-deploy

>       ```

Reviewer 2: These should be in the same code block. This is an example of how the commands should be run one after another in a script.

      Alternatively, if you use `docker-compose` commands directly, run the `docker-compose run deploy cloud-post-deploy` command after the deploy command.

-  **Refreshing the database**–The Database container is now stored in a persistent Docker volume named `magento-db`. When you refresh the Docker environment, the database is no longer deleted automatically. If needed, use one of the following commands to manually remove it.

   -  Remove the `magento-db` container:

      ```bash
      docker volume rm magento-db
      ```

>    -  Remove the `magento-db` volume and all associated volumes when you shut down the containers:

Reviewer 2: `docker-compose down -v` Shuts down all the Docker containers associated with docker-compose.yml and then the `-v` switch removes the associated volumes.
```suggestion
   -  Remove all associated volumes when shutting down the Docker containers:
```

      ```bash

>       docker-compose -v

Reviewer 2: ```suggestion
      docker-compose down -v
```
The intended command here is `docker-compose down -v`.

      ```

> -  **Override file synchronization settings for archive and backup files**–Archive and backup files with the following extensions are no longer synchronized when using docker-sync or mutagen:  `*.sql`, `*.gz`, `*.zip`, and `*.bz2`.  You can override the default file synchronization for these file types by renaming the file to end with a different extension, for example `synchronize-me.zip-backup`.

Reviewer: > for example `synchronize-me.zip-backup`.

This appears as if the period is part of the filename. It is tricky to end a sentence with something like this, and in this paragraph you have two. Consider:

```suggestion
-  **Override file synchronization settings for archive and backup files**–Archive and backup files with the following extensions are no longer synchronized when using docker-sync or mutagen:  `*.sql`, `*.gz`, `*.zip`, and `*.bz2`. You can override the default file synchronization for these file types by renaming the file to end with a different extension. For example: `synchronize-me.zip-backup`
```
